src/pages/utilities/FileSearch.tsx

```diff
@@ -0,0 +1,446 @@
+  });
+
+  const rehashFiles = useEventCallback(() => {
+    setSelectedRows([]);
+    let failedFiles = 0;
+
+    forEach(selectedRows, (file) => {
+      rehashFile(file.ID).catch((error) => {
+        failedFiles += 1;
+        console.error(error);
+      });
+    });
+
+    if (failedFiles) toast.error(`Rehash failed for ${failedFiles} files!`);
+  });
+
+  const rescanFiles = useEventCallback(() => {
+    setSelectedRows([]);
+    let failedFiles = 0;
+    forEach(selectedRows, (file) => {
+      rescanFile(file.ID).catch((error) => {
+        failedFiles += 1;
+        console.error(error);
+      });
+    });
+
+    if (failedFiles) toast.error(`Rescan failed for ${failedFiles} files!`);
+  });
+
+  return (
+    <div className="relative box-border flex grow items-center rounded-md border border-panel-border bg-panel-background-alt px-4 py-3">
+      <TransitionDiv className="absolute flex grow gap-x-4" show={selectedRows.length === 0}>
+        <MenuButton
+          onClick={() => {
+            setSelectedRows([]);
+            invalidateQueries(['files', { include_only: ['Unrecognized'] }]);
+          }}
+          icon={mdiRefresh}
+          name="Refresh"
+        />
+      </TransitionDiv>
+      <TransitionDiv className="absolute flex grow gap-x-4" show={selectedRows.length !== 0}>
+        <MenuButton onClick={() => rescanFiles()} icon={mdiDatabaseSearchOutline} name="Rescan" />
+        <MenuButton onClick={() => rehashFiles()} icon={mdiDatabaseSyncOutline} name="Rehash" />
+        <MenuButton onClick={showDeleteConfirmation} icon={mdiMinusCircleOutline} name="Delete" highlight />
+        <MenuButton
+          onClick={() => setSelectedRows([])}
+          icon={mdiCloseCircleOutline}
+          name="Cancel Selection"
+          highlight
+        />
+      </TransitionDiv>
+      <span className="ml-auto font-semibold text-panel-text-important">
+        {selectedRows.length}
+        &nbsp;
+      </span>
+      {selectedRows.length === 1 ? 'File ' : 'Files '}
+      Selected
+      <DeleteFilesModal
+        show={showConfirmModal}
+        selectedFiles={selectedRows}
+        removeFile={removeFileFromSelection}
+        onClose={cancelDelete}
+        onConfirm={deleteFiles}
+      />
+    </div>
+  );
+};
+
+const FileSelected = (props: FileSelectedProps) => {
+  const { fileId } = props;
+  const { data: file } = useFileQuery(fileId, { include: ['XRefs', 'MediaInfo'] }, !!fileId);
+  const fileAniDbUrl = `https://anidb.net/file/${file?.AniDB?.ID}`;
+  const seriesShokoId = useMemo(() => get(file, 'SeriesIDs.0.SeriesID.ID', 0), [file]);
+  const { data: seriesInfo } = useSeriesAniDBQuery(
+    get(file, 'SeriesIDs.0.SeriesID.AniDB', 0),
+    get(file, 'SeriesIDs.0.SeriesID.AniDB', 0) !== 0,
+  );
+  const { data: episodeInfo } = useEpisodeAniDBQuery(
+    get(file, 'SeriesIDs.0.EpisodeIDs.0.ID', 0),
+    get(file, 'SeriesIDs.0.EpisodeIDs.0.ID', 0) !== 0,
+  );
```

Comment:
```suggestion
  const seriesAnidbId = file?.SeriesIDs?.[0].SeriesID.AniDB;
  const episodeId = file?.SeriesIDs?.[0].EpisodeIDs[0].ID;
  const { data: seriesInfo } = useSeriesAniDBQuery(seriesAnidbId!, !!seriesAnidbId);
  const { data: episodeInfo } = useEpisodeAniDBQuery(episodeId!, !!episodeId);
```

Reply:
More readable and less duplicate code

Reply:
Also, need to change the type of `EpisodeIDs` to `EpisodeIDsType[]` in `XRefsType`. It seems to be wrong
![image](https://github.com/ShokoAnime/Shoko-WebUI/assets/26010946/cb106876-d9f8-47f2-b1ea-1afab4c63535)